File: articles/causal-inference-and-computational-reasoning/typescript/causal_effect.ts
type Contrast = {
  treatedMean: number;
  controlMean: number;
};

function effect(contrast: Contrast): number {
  return contrast.treatedMean - contrast.controlMean;
}

console.log(`causal contrast = ${effect({ treatedMean: 0.64, controlMean: 0.47 }).toFixed(4)}`);

File: articles/causal-inference-and-computational-reasoning/haskell/CausalEffect.hs
module Main where

data Contrast = Contrast { treatedMean :: Double, controlMean :: Double }

effect :: Contrast -> Double
effect c = treatedMean c - controlMean c

main :: IO ()
main = do
  let c = Contrast 0.64 0.47
  putStrLn ("causal contrast = " ++ show (effect c))
